internal/metamorph/processor.go

```diff
@@ -582,69 +583,57 @@ func (p *Processor) StartRequestingSeenOnNetworkTxs() {
 	}()
 }
 
+// Periodically read transactions with status lower then SEEN_ON_NETWORK from database and announce them again
 func (p *Processor) StartProcessExpiredTransactions() {
-	ticker := time.NewTicker(p.processExpiredTxsInterval)
 	p.waitGroup.Add(1)
 
 	go func() {
 		defer p.waitGroup.Done()
+
+		ticker := time.NewTicker(p.processExpiredTxsInterval)
+		defer ticker.Stop()
+
+		const (
+			buffSize  = 512
+			maxBufCap = 1024
+		)
+
+		toAnnounceBuffer := make([]*chainhash.Hash, 0, buffSize)
+		toRequestBuffer := make([]*chainhash.Hash, 0, buffSize)
+
 		for {
 			select {
 			case <-p.ctx.Done():
 				return
-			case <-ticker.C: // Periodically read unmined transactions from database and announce them again
+			case <-ticker.C:
 				ctx, span := tracing.StartTracing(p.ctx, "StartProcessExpiredTransactions", p.tracingEnabled, p.tracingAttributes...)
 
 				// define from what point in time we are interested in unmined transactions
 				getUnminedSince := p.now().Add(-1 * p.mapExpiryTime)
-				var offset int64
 
 				requested := 0
 				announced := 0
-				for {
-					// get all transactions since then chunk by chunk
-					unminedTxs, err := p.store.GetUnmined(ctx, getUnminedSince, loadUnminedLimit, offset)
-					if err != nil {
-						p.logger.Error("Failed to get unmined transactions", slog.String("err", err.Error()))
-						break
-					}
 
-					offset += loadUnminedLimit
-					if len(unminedTxs) == 0 {
+				token := newContinuationToken(getUnminedSince, loadUnminedLimit)
+				for token.next {
+					token, toRequestBuffer, toAnnounceBuffer = p.getTxsToReprocess(ctx, token, toRequestBuffer, toAnnounceBuffer)
+					if token.err != nil {
+						p.logger.Error("Failed to get unmined transactions", slog.Time("since", getUnminedSince), slog.String("err", token.err.Error()))
```

Comment:
Is this refactoring related to the change of p2p package in metamorph? It looks like it is independent from that.

Could this be moved to a separate PR? This one is already quite big.

Reply:
Initially, this change was necessary because I hadn't included batching in the new p2p.

Reverted